minecraft/service/signaling/messaging/dial.go

```
@@ -0,0 +1,160 @@
+	token, err := src.ServiceToken(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("request service token: %w", err)
+	}
+
+	opts := &websocket.DialOptions{
+		HTTPHeader: make(http.Header),
+		HTTPClient: d.HTTPClient,
+	}
+	opts.HTTPHeader.Set("Authorization", token.AuthorizationHeader)
+	requestURL := d.Environment.ServiceURI.JoinPath("/ws/v1.0/messaging/connect")
+	c, _, err := websocket.Dial(ctx, requestURL.String(), opts)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)